lib/Dialect/FIRRTL/Transforms/ExtractInstances.cpp

```diff
@@ -1011,7 +1034,10 @@ void ExtractInstancesPass::createTraceFiles() {
   // Generate the trace files.
   SmallVector<Attribute> symbols;
   SmallDenseMap<Attribute, unsigned> symbolIndices;
+  if (sifiveMetadataClass && !extractMetadataClass)
+    createSchema();
```

Comment:
Just want to be sure--createSchema and createPathRef might modify the top-level builtin.module, so is this happening in a single thread?

Reply:
Right, this is all in a single thread.